src/dreyfus_index.erl

```diff
@@ -157,9 +159,28 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult}, State) ->
+    #state{index=#index{sig=CurrentSig}, waiting_list = WaitList} = State,
+    case check_if_index_is_deleted(DDocResult, CurrentSig) of
+        true ->
+            case WaitList of
+                [] ->
+                    self() ! index_deleted;
+                [_Index] ->
+                    erlang:send_after(?INDEX_SHUTDOWN_DELAY, self(), index_deleted)
```

Comment:
I thought the whole idea of this work is to stop indexing for the deleted indices which could have already triggered and in progress. Agree that there won't be any _search end point requests after the index is deleted. Basically giving 1 minute to complete the pending requests, if they are not done then it will end the request.